tests/test_app/tests.py

```diff
@@ -1,12 +1,144 @@
+    """Tests for the SocialTokenObtainSerializer serializer."""
+
+    def setUp(self):
+        """Set up the test case by creating necessary test data and objects."""
+        self.user = User.objects.create_user(
+            email="testuser@example.com", password="testpassword"
+        )
+        # Create request object
+        self.request = APIRequestFactory()
+        self.request.user = self.user
+        self.request.session = {}
+        self.request.social_strategy = DRFStrategy(DjangoStorage, self.request)
+        self.access_token = "test"
+
+        self.serializer = SocialTokenObtainSerializer
+
+    def test_fields(self):
+        """Test field requirement."""
+        required_fields = ["provider", "access_token"]
+        not_required_fields = [
+            "response",
+        ]
+        for field in required_fields:
+            self.assertTrue(self.serializer().fields[field].required)
+
+        for field in not_required_fields:
+            self.assertFalse(self.serializer().fields[field].required)
+
+    @patch("df_auth.drf.serializers.load_backend")
+    def test_valid_data_oauth2(self, load_backend_mock):
+        """Test that a valid social access token can be used to authenticate a user."""
+        provider = "facebook"
+
+        # Mock return values.
+        backend = MagicMock()
+        backend.name = provider
+        load_backend_mock.return_value = backend
+        mock_do_auth = backend.do_auth = MagicMock(return_value=self.user)
+
+        valid_data = {
+            "provider": "facebook",
+            "access_token": self.access_token,
+            "first_name": "John",
+            "last_name": "Doe",
+        }
+        serializer = self.serializer(data=valid_data, context={"request": self.request})
+        # Test to ensure no exception was raised.
+        serializer.is_valid(raise_exception=True)
+
+        # Ensure the mocked methods are called with the correspecting parameters.
+        mock_do_auth.assert_called_once_with(self.access_token, user=self.user)
+        load_backend_mock.assert_called_once_with(
+            self.request.social_strategy, provider, redirect_uri=None
+        )
+
+        # Ensure the `token` key exist in the validated response
+        self.assertIn("token", serializer.validated_data)
+
+
+class SocialOAuth1TokenObtainSerializerTests(APITestCase):
+    """Tests for the SocialTokenObtainSerializer serializer."""
+
+    def setUp(self):
+        """Set up the test case by creating necessary test data and objects."""
+        self.user = User.objects.create_user(
+            email="testuser@example.com", password="testpassword"
+        )
+        # Create request object
+        self.request = APIRequestFactory()
+        self.request.user = self.user
+        self.request.session = {}
+        self.request.social_strategy = DRFStrategy(DjangoStorage, self.request)
+
+        self.serializer = SocialOAuth1TokenObtainSerializer
+
+    def test_fields(self):
```

Comment:
This test seems useless for me. It checks not the logic but the declarations of SocialTokenObtainSerializer